.github/workflows/api-ci.yml

```
@@ -85,7 +85,55 @@ jobs:
           fi
 
+      - name: Verify VERCEL_AUTOMATION_BYPASS_SECRET for E2E
+        run: |
+          if [ -z "${{ secrets.VERCEL_AUTOMATION_BYPASS_SECRET }}" ]; then
+            echo "::error::VERCEL_AUTOMATION_BYPASS_SECRET secret is required for E2E against Vercel previews. Add it in repository secrets."
+            exit 1
+          fi
+
+      - name: Get branch ref for Vercel deployment URL
+        id: vercel-url
+        run: |
```

actionlint (1.7.10)

[error] "github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)